sql/expression/comparison.go

```diff
@@ -236,25 +259,25 @@ func (lt *LessThan) TransformUp(f func(sql.Expression) (sql.Expression, error))
 // GreaterThanOrEqual is a comparison that checks an expression is greater or equal to
 // another.
 type GreaterThanOrEqual struct {
-	Comparison
+	comparison
 }
 
 // NewGreaterThanOrEqual creates a new GreaterThanOrEqual
 func NewGreaterThanOrEqual(left sql.Expression, right sql.Expression) *GreaterThanOrEqual {
-	return &GreaterThanOrEqual{NewComparison(left, right)}
+	return &GreaterThanOrEqual{newComparison(left, right)}
 }
 
 // Eval implements the Expression interface.
 func (gte GreaterThanOrEqual) Eval(
 	session sql.Session,
 	row sql.Row,
 ) (interface{}, error) {
-	a, err := gte.Left.Eval(session, row)
+	a, err := gte.BinaryExpression.Left.Eval(session, row)
```

Comment:
why add the `BinaryExpression` here?

Reply:
Because of the `Left()` method in the interface, it shadows the `Left` field of the embedded `BinaryExpression`.